### PR TITLE
feat: Announce form field errors using ARIA live region

### PR DESCRIPTION
**Pull request by cansuaa**
### Description

Added aria live to form field error messages as part of validation improvements. These changes allow error messages to be announced in form field, attribute editor, tag editor, file upload, etc (components that use FormFieldError component).

Issue: AWSDesignSystem-432

### How has this been tested?

Tested manually (to try yourself; go to pages/form/with-validation, turn on screen reader, type text longer than 4 letters to the input field and click submit)
Also added and ran unit test

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/form-field/internal.tsx [hunk 1]
```diff
@@ -38,18 +39,23 @@ export function FormFieldError({ id, children, errorIconAriaLabel }: FormFieldEr
...
+            <InternalIcon name="status-warning" size="small" />
+          </div>
         </div>
+        <span className={styles.error__message}>{children}</span>
       </div>
-      <span className={styles.error__message}>{children}</span>
-    </div>
+      <LiveRegion assertive={true}>
+        {errorIconAriaLabel} {children}
+      </LiveRegion>
```
Comment (michaeldowseza) [inline src/form-field/internal.tsx:57]: Should we consider wrapping both of these in `spans` as per the comments here in the LiveRegion implementation?

https://github.com/cloudscape-design/components/blob/main/src/internal/components/live-region/index.tsx#L37-L48

**Comment by michaeldowseza** [comment]
What happens when you end up with multiple errors at the same time? Would we just be flooding the screen reader? Does the error message that is read out include the FormField label?

File: src/form-field/internal.tsx (hunk 1 above)
Comment (cansuaa) [reply src/form-field/internal.tsx:57]: Sure, will update it.

**Comment by cansuaa** [comment]
> What happens when you end up with multiple errors at the same time? Would we just be flooding the screen reader? Does the error message that is read out include the FormField label?

I tested with multiple form fields which I passed errorText to all at the same time. It only announces one of them, usually the first one. It doesn't read out the form field label

File: src/form-field/internal.tsx [hunk 2]
```diff
@@ -38,18 +39,24 @@ export function FormFieldError({ id, children, errorIconAriaLabel }: FormFieldEr
...
+        <div className={styles['error-icon-shake-wrapper']}>
+          <div
+            role="img"
+            aria-label={i18n('i18nStrings.errorIconAriaLabel', errorIconAriaLabel)}
+            className={styles['error-icon-scale-wrapper']}
+          >
+            <InternalIcon name="status-warning" size="small" />
+          </div>
         </div>
+        <span className={styles.error__message}>{children}</span>
```
[the remark this answers is not in the document]
Comment (pan-kot) [reply src/form-field/internal.tsx:53]: We can wrap the entire thing with `<LiveRegion visible={true} />`. The simple aria-live used to have a bug in Firefox.

**Comment by pan-kot** [comment]
> I tested with multiple form fields which I passed errorText to all at the same time. It only announces one of them, usually the first one. It doesn't read out the form field label

Did you test it with different browsers and screen-readers? I would actually expect multiple errors to get announced. I think the reason only one get announced is because you set the live region type to "assertive". The assertive live regions tend to overshadow one another and I am not sure it is the first one that consistently wins.

**Comment by cansuaa** [comment]
> > I tested with multiple form fields which I passed errorText to all at the same time. It only announces one of them, usually the first one. It doesn't read out the form field label
> 
> Did you test it with different browsers and screen-readers? I would actually expect multiple errors to get announced. I think the reason only one get announced is because you set the live region type to "assertive". The assertive live regions tend to overshadow one another and I am not sure it is the first one that consistently wins.

Not with different browser but I tried with VoiceOver and NVDA. The case you mentioned is also quite possible. I'm currently testing out different implementations and will discuss them with accessibility.

**Review by timogasda — APPROVED** [review]

**Review by michaeldowseza — APPROVED** [review]

**Comment by timogasda** [comment]
Successful internal dryrun: `6813817202`

**Pull request merged**